test/functional/tests/cache_ops/test_concurrent_cores_flush.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+from time import sleep
+import pytest
+
+from api.cas import casadm, casadm_parser, cli
+from api.cas.cache_config import CacheMode, CleaningPolicy, CacheModeTrait
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from core.test_run import TestRun
+from test_tools.dd import Dd
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+cache_size = Size(4, Unit.GibiByte)
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_concurrent_cores_flush(cache_mode):
+    """
+        title: Test if OpenCAS works correctly when concurrent cores makes flushing simultaneously.
+        description: |
+          Concurrent cores of shared cache should be able to flush dirty data simultaneously
+          and independently.
+        pass_criteria:
+          - No system crash.
+          - Flushing of both cores would be finished nith success.
+    """
+    with TestRun.step("Prepare cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([cache_size])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([cache_size * 3] * 2)
+        core_part1 = core_dev.partitions[0]
+        core_part2 = core_dev.partitions[1]
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+    with TestRun.step("Set cleaning policy to NOP."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step(f"Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+
+    with TestRun.step("Put workload on concurrent cores."):
+        bs = Size(4, Unit.MebiByte)
+        cnt = int(cache_size.value / bs.value)
+
+        Dd().output(core1.system_path) \
+            .input("/dev/urandom") \
+            .block_size(bs) \
+            .count(cnt) \
+            .run_in_background()
+
+        Dd().output(core2.system_path) \
+            .input("/dev/urandom") \
+            .block_size(bs) \
+            .count(cnt) \
+            .run()
```

Comment:
After second `dd` whole cache could be occupied by data only from one core.  Writing half of cache size to each core would be more reliable.